papaparse.js

```diff
@@ -1457,6 +1457,32 @@ License: MIT
 			if (!input)
 				return returnable();
 
+			// Rename headers if there are duplicates
+			if (config.header)
+			{
+				let firstLine = input.split(newline)[0]
+				let headers = firstLine.split(delim) 
+				let separator = '_';
+				let header_map = []; 
+				let header_count = {};
+
+				for (var i in headers){
+					var header = headers[i]
+					if (isFunction(config.transformHeader))
+						header = config.transformHeader(header, i);
+					var header_name = header
+
+					let count = header_count[header] || 0
+					if (count > 0) header_name = header+separator+count;
+					header_count[header] = count+1
+
+					header_map.push(header_name)
+				}
+				var edited_input = input.split(newline)
```

Comment:
We should avoid spliting the input when the computed header is the same as the previous one. 
This will avoid some performance degradation on bigger files without duplicated headers.

So I think we should: 

1. Get the first line (you already have it as `headers` variable)
2. Compute the new header: `header_map.join(delim)`
3. If both values are different alter the input (current code). Otherwise do nothing.

Reply:
Great catch. I've updated this now. It will still fire if transformHeaders is called. This can be fixed by setting a flag i.e. `edited = true` in:

```
if (count > 0) {
	headerName = header + separator + count;
}
```

and then checking for that flag instead of comparing the two arrays. Otherwise it works fine. Let me know if you'd rather this solution.